### ISSUE TITLE
Fix Request: false positive in AvoidSideEffectsInStreams

### ISSUE THREAD

**Issue by jborgers** [jborgers/PMD-jPinpoint-rules#627]
```
    private void handleCompletionErrors(List<String> errors) {
        List<ErrorDetail> errorDetails = errors.stream() // <-- false positive
            .map(error -> {
                ErrorDetail errorDetail = new ErrorDetail();
                errorDetail.setErrorCode(error);
                return errorDetail;
            })
            .toList();
        processItem.getErrors().addAll(errorDetails);
    }
```
Note that ErrorDetail cannot be resolved.

**Comment by jborgers** [comment]
The setter on errorDetail triggers it. Since that is a local var, it is not a side effect.
